rmw_zenoh_cpp/CMakeLists.txt

```diff
@@ -21,9 +21,15 @@ find_package(rcutils REQUIRED)
 find_package(rosidl_typesupport_fastrtps_c REQUIRED)
 find_package(rosidl_typesupport_fastrtps_cpp REQUIRED)
 find_package(rmw REQUIRED)
+find_package(rmw_dds_common REQUIRED)
 find_package(tracetools REQUIRED)
 find_package(zenoh_cpp_vendor REQUIRED)
 
+if(SECURITY)
+  find_package(OpenSSL REQUIRED)
+  set(HAVE_SECURITY 1)
```

Comment:
Let's not have any special build flags for security. We will overwrite the Zenoh config with auth params if `security_options` is present in `rmw_context_impl_s`.